wagtail/search/backends/database/mysql/mysql.py

```diff
@@ -610,7 +618,7 @@ def finish(self):
 class MySQLSearchAtomicRebuilder(MySQLSearchRebuilder):
     def __init__(self, index):
         super().__init__(index)
-        self.transaction = transaction.atomic(using=index.db_alias)
+        self.transaction = transaction.atomic(using=index.read_connection.alias)
```

Comment:
The rebuilder class is populating the index table, so shouldn't this be `write_connection`? And in fact this seems to be the only place where `read_connection` is used, so maybe we don't need `read_connection` at all?